ecosystem/sep-0006.md

```diff
@@ -575,6 +850,70 @@ The response should be a JSON object like:
       "enabled": false
     }
   },
+  "withdraw_indicative_quote": {
+    "USD": {
+      "enabled": true,
+      "authentication_required": true,
+      "fee_fixed": 5,
+      "fee_percent": 0,
```

Comment:
> I think we need to mention that these attributes should only be present if fees are charged in units of the asset on Stellar

Only for `withdraw*_quotes` or also for `deposit*_quotes`? I did it for withdraws in https://github.com/stellar/stellar-protocol/pull/973/commits/0ae235ddd62fb7e5f4bb30f26bc4b24d13fbb41a.

Reply:
> And, we should add `source_asset` & `destination_asset` as parameters to `GET /fee` as well.
> 
> Instead of adding these parameters to `/fee` we could create new endpoints `/fee/indicative_quote` and `fee/firm_quote` too.

If we make it so fees are always charged in Stellar assets we shouldn't need these changes. But maybe that is not a real-world assumption 🤔 . WDYT?

Reply:
> Only for withdraw*_quotes or also for deposit*_quotes? I did it for withdraws in 0ae235d.

For both, since fees could be charged in either currency for deposits as well.

> If we make it so fees are always charged in Stellar assets we shouldn't need these changes

Thats true, but I think its likely that businesses will want to charge fees in terms of the off-chain asset. We could make the argument that since the entity charging deposit/withdrawal fees is the same entity doing the FX, the entity is capable of charging fees in terms of the on-chain asset while designing the UX to communicate fees in terms of the off-chain asset. However I don't think we should introduce this complication if our protocol can support it with changes.

Reply:
> Since the fees charged can be in units of either asset, I think we need to mention that these attributes should only be present if fees are charged in units of the asset on Stellar, since it would be misleading if fees were charged in terms of the off-chain asset.

Ok, now I'm making it clear that fees should be charged in terms of the incoming assets:
* withdrawal fees are charged in units of the **on-chain** assets: https://github.com/stellar/stellar-protocol/commit/0ae235ddd62fb7e5f4bb30f26bc4b24d13fbb41a
* deposits fees are charged in units of **off-chain** assets: https://github.com/stellar/stellar-protocol/pull/973/commits/12a2e4d27a9847e3dd57a7ee68a274fe2f8372c8

Also, added the new fee endpoints in https://github.com/stellar/stellar-protocol/pull/973/commits/373a083ab58b5044da385c66b230ac83119dafc2.